azurerm/internal/services/resource/tests/resource_arm_provider_registration_test.go

```diff
@@ -0,0 +1,99 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMResourceProviderRegistration_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_resource_provider_registration", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMResourceProviderDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMResourceProviderRegistration_basic(),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMResourceProviderRegistrationExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "resource_provider_namespace"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMResourceProviderRegistrationExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Resource.ProvidersClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %q", resourceName)
+		}
+
+		resourceProviderNamespace := rs.Primary.Attributes["resource_provider_namespace"]
+
+		resp, err := client.Get(ctx, resourceProviderNamespace, "")
+
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Resource Provider Namespace %q is not found", resourceProviderNamespace)
+			}
+			return fmt.Errorf("Bad: Get on ProvidersClient: %+v", err)
+		}
+
+		if resp.RegistrationState != nil && *resp.RegistrationState != "Registered" {
+			return fmt.Errorf("Bad: Resource Provider Namespace %q is not registered", resourceProviderNamespace)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMResourceProviderDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Resource.ProvidersClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_resource_provider_registration" {
+			continue
+		}
+
+		resourceProviderNamespace := rs.Primary.Attributes["resource_provider_namespace"]
+
+		resp, err := client.Get(ctx, resourceProviderNamespace, "")
+
+		if err != nil {
+			return err
+		}
+
+		if resp.RegistrationState != nil && *resp.RegistrationState != "Unregistered" {
+			return fmt.Errorf("Bad: Resource Provider Namespace %q is not unregistered", resourceProviderNamespace)
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMResourceProviderRegistration_basic() string {
+	return (`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_provider_registration" "test" {
+  resource_provider_namespace = "Microsoft.PolicyInsights"
```

Comment:
(as above) this wants to be `name`:

```suggestion
  name = "Microsoft.PolicyInsights"
```